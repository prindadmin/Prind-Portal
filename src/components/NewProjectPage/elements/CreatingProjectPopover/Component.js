import React, { Component } from 'react'
//import PropTypes from 'prop-types'

import PopOverHandler from '../../../common/popOverHandler'

import {
  Spinner,
  Intent,
} from '@blueprintjs/core'

import * as strings from '../../../../data/Strings'

export class Element extends Component {
  static propTypes = {
  }

  getContent = () => {
    return (
      <React.Fragment>
        {strings.PROJECT_CREATION_BILLING_WARNING}
        <Spinner size='150' intent={Intent.DANGER}/>
      </React.Fragment>
    )
  }

  render () {
    return (
      <PopOverHandler>
        <div id='popup-greyer' onClick={(e) => e.stopPropagation()}>
          <div id='project-creation-popover'>
            <div id='popup-box'>
              <div className='project-creation-popover-container' onClick={(e) => e.stopPropagation()}>
                <div className='element-title'>
                  {strings.PROJECT_CREATION_IN_PROGRESS}
                </div>
                <div className='element-description'>
                  {this.getContent()}
                </div>
              </div>
            </div>
          </div>
        </div>
      </PopOverHandler>
    )
  }

}

export default Element
