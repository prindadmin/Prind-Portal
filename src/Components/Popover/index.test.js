
import Component from './index'
import { mount } from 'enzyme';

import { JSDOM } from "jsdom"

var props;

beforeAll(() => {
  document.body.innerHTML =
    '<div id="root">' +
    '</div>' +
    '<div id="popOver">' +
    '</div>';
});

beforeEach(() => {
  props = {
    children: [<div key='1' id='hello-dom' />],
    handleClose: jest.fn()
  }
});

it('Popover - Should render', () => {
  const component = mount(<Component {...props} />);
  expect(component).toMatchSnapshot();
});

it('Popover - Should click container and do nothing', () => {
  const mockStopPropagation = jest.fn()
  const component = mount(<Component {...props} />);
  const container = component.find('#popover-container')
  container.simulate('click', { stopPropagation: mockStopPropagation })
  expect(mockStopPropagation).toHaveBeenCalledTimes(2);
});

it('Popover - Should click greyer and close', () => {
  const mockStopPropagation = jest.fn()
  const component = mount(<Component {...props} />);
  const greyer = component.find('#popover-greyer')
  greyer.simulate('click', { stopPropagation: mockStopPropagation })
  expect(mockStopPropagation).toHaveBeenCalledTimes(1);
  expect(props.handleClose).toHaveBeenCalledTimes(1);
});
