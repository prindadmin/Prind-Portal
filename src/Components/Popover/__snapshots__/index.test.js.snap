// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover - Should render 1`] = `
<Popover
  handleClose={[MockFunction]}
>
  <PopOverPortal>
    <Portal
      containerInfo={
        <div
          id="popOver"
        >
          <div
            id="popover-greyer"
          >
            <div
              id="popover-container"
            >
              <div
                id="hello-dom"
              />
            </div>
          </div>
        </div>
      }
    >
      <div
        id="popover-greyer"
        onClick={[Function]}
      >
        <div
          id="popover-container"
          onClick={[Function]}
        >
          <div
            id="hello-dom"
            key="1"
          />
        </div>
      </div>
    </Portal>
  </PopOverPortal>
</Popover>
`;
